src/test/scala/com/codacy/tools/spotbugs/helper/LanguageHelperSpec.scala

```
@@ -0,0 +1,50 @@
+package com.codacy.tools.spotbugs.helper
+
+import java.nio.file.Paths
+
+import com.codacy.tool.spotbugs.helper.LanguageHelper
+import org.scalatest.{Matchers, WordSpecLike}
+
+class LanguageHelperSpec extends Matchers with WordSpecLike {
+
+  "LanguageHelper" when {
+    "isScala" should {
+
+      "detect scala on a set of scala files" in {
+        val files = Option(
+          Set(
+            Paths.get("bla/bla/bleeeee/Carlos.scala"),
+            Paths.get("blaaaa/ble/Carlos2.scala"),
+            Paths.get("path/to/file/Carlos1000.scala")
+          )
+        )
+
+        LanguageHelper.isScala(files) shouldBe true
+      }
+
+      "not detect scala on a set of non-scala files" in {
+        val files = Option(
+          Set(
+            Paths.get("bla/bla/bleeeee/Carlos.java"),
+            Paths.get("blaaaa/ble/Carlos2.py"),
+            Paths.get("path/to/file/Carlos1000.notalanguage")
+          )
+        )
+
+        LanguageHelper.isScala(files) shouldBe false
+      }
+
+      "not detect scala on a set with a mix of scala and non-scala files" in {
```

uhmmm, this is a tricky decision, in scala projects usually you can just drop a java file in the middle and everything will work as expected, I would fail hard in this case to warn the user about what is going on.

---

- The product will never send mixed language files 
- The exception is custom extensions, in that case we are keeping the old behaviour of the tool to make sure we dont break the java analysis (checked with product team to confirm this, got a go-ahead)